### PR TITLE
Fix null checks per LGTM

### PR DESCRIPTION
**Pull request by sixlettervariables**
I went through our active list of null check alerts on LGTM and fixed them in one of three ways:
1. Removed a useless null check.
2. Added a needed null check.
3. Reorganized the code to help the analyzer with null detection.

https://lgtm.com/projects/g/MegaMek/mekhq/alerts/?mode=list&severity=

One commit per file.

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/parts/MissingPart.java [hunk 1]
```diff
@@ -369,12 +369,15 @@ public String checkScrappable() {
...
 			replace.decrementQuantity();
-		}
+            return replace.getName() + " scrapped.";
+        }
+
 		skillMin = SkillType.EXP_GREEN;
-		return replace.getName() + " scrapped.";
+
+        return getName() + " scrapped.";
 	}
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/MissingPart.java:381]: Looks like this is tab spaced.

Comment (sixlettervariables) [reply MekHQ/src/mekhq/campaign/parts/MissingPart.java:381]: I made no attempt to adjust whitespacing in files.

File: MekHQ/src/mekhq/gui/model/FilterableListModel.java [hunk 2]
```diff
@@ -119,20 +119,20 @@ public void updateFilter() {
...
     public int getSize() {
         return (null == filter) ? ((null == peerModel) ? 0 : peerModel.getSize()) : indices.size();
     }
 
     @Override
     public E getElementAt(int index) {
-        if(null == filter) {
-            if(null != peerModel) {
+        if(null != peerModel) {
+            if(null == filter) {
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/model/FilterableListModel.java:131]:
```suggestion
            if (null == filter) {
```

File: MekHQ/src/mekhq/gui/model/FilterableListModel.java [hunk 3]
```diff
@@ -119,20 +119,20 @@ public void updateFilter() {
...
     @Override
     public int getSize() {
         return (null == filter) ? ((null == peerModel) ? 0 : peerModel.getSize()) : indices.size();
     }
 
     @Override
     public E getElementAt(int index) {
-        if(null == filter) {
-            if(null != peerModel) {
+        if(null != peerModel) {
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/model/FilterableListModel.java:130]:
```suggestion
        if (null != peerModel) {
```

File: MekHQ/src/mekhq/campaign/parts/Part.java [hunk 4]
```diff
@@ -980,21 +980,22 @@ public TargetRoll getAllModsForMaintenance() {
         TargetRoll mods = new TargetRoll(campaign.getCampaignOptions().getMaintenanceBonus(), "maintenance");
         mods.addModifier(Availability.getTechModifier(getTechRating()), "tech rating " + ITechnology.getRatingName(getTechRating()));
 
-        if(null != unit) {
-            mods.append(unit.getSiteMod());
-            if(unit.getEntity().hasQuirk("easy_maintain")) {
+        if(null != getUnit()) {
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/Part.java:983]:
```suggestion
        if (null != getUnit()) {
```

File: MekHQ/src/mekhq/campaign/parts/Part.java [hunk 5]
```diff
@@ -980,21 +980,22 @@ public TargetRoll getAllModsForMaintenance() {
...
+            mods.append(getUnit().getSiteMod());
+            if(getUnit().getEntity().hasQuirk("easy_maintain")) {
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/Part.java:985]:
```suggestion
            if (getUnit().getEntity().hasQuirk("easy_maintain")) {
```

File: MekHQ/src/mekhq/campaign/parts/Part.java [hunk 6]
```diff
@@ -980,21 +980,22 @@ public TargetRoll getAllModsForMaintenance() {
...
                 mods.addModifier(-1, "easy to maintain");
             }
-            else if(unit.getEntity().hasQuirk("difficult_maintain")) {
+            else if(getUnit().getEntity().hasQuirk("difficult_maintain")) {
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/Part.java:988]:
```suggestion
            else if (getUnit().getEntity().hasQuirk("difficult_maintain")) {
```

File: MekHQ/src/mekhq/campaign/parts/Part.java [hunk 7]
```diff
@@ -980,21 +980,22 @@ public TargetRoll getAllModsForMaintenance() {
...
             }
-        }
-        if(isClanTechBase() || (this instanceof MekLocation && this.getUnit() != null && this.getUnit().getEntity().isClan())) {
-            if (unit.getTech() == null) {
-                mods.addModifier(2, "Clan tech");
-            } else if (!unit.getTech().isClanner()
-                    && !unit.getTech().getOptions().booleanOption(PersonnelOptions.TECH_CLAN_TECH_KNOWLEDGE)) {
-                mods.addModifier(2, "Clan tech");
+
+            if(isClanTechBase() || (this instanceof MekLocation && getUnit().getEntity().isClan())) {
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/parts/Part.java:992]:
```suggestion
            if (isClanTechBase() || ((this instanceof MekLocation) && getUnit().getEntity().isClan())) {
```

**Pull request merged**